src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -899,53 +897,49 @@ private static void SetPathAndQueryParameters(CodeMethod target, OpenApiUrlTreeN
     {
-        var type = GetPrimitiveType(schema?.Items, string.Empty);
+        CodeTypeBase type = GetPrimitiveType(schema?.Items, string.Empty);
         if (type == null || string.IsNullOrEmpty(type.Name))
         {
             var targetNamespace = schema?.Items == null ? codeNamespace : GetShortestNamespace(codeNamespace, schema.Items);
-            type = CreateModelDeclarationAndType(currentNode, schema?.Items, operation, targetNamespace);
+            type = CreateModelDeclarations(currentNode, schema?.Items, operation, targetNamespace, default , typeNameForInlineSchema: typeNameForInlineSchema);
         }
         type.CollectionKind = CodeTypeBase.CodeTypeCollectionKind.Complex;
         return type;
     }
-    private CodeElement GetExistingDeclaration(bool checkInAllNamespaces, CodeNamespace currentNamespace, OpenApiUrlTreeNode currentNode, string declarationName) {
-        var localNameSpace = GetSearchNamespace(false, currentNode, currentNamespace);
-        var localItemSearchItem = localNameSpace.FindChildByName<ITypeDefinition>(declarationName, checkInAllNamespaces) as CodeElement;
-        if (!checkInAllNamespaces || localItemSearchItem != null)
-            return localItemSearchItem; // if we can find an item in the target namespace lets default to that.
-
-        var globalSearchNameSpace = GetSearchNamespace(checkInAllNamespaces, currentNode, currentNamespace);
-        return globalSearchNameSpace.FindChildByName<ITypeDefinition>(declarationName, checkInAllNamespaces) as CodeElement;
+    private CodeElement GetExistingDeclaration(CodeNamespace currentNamespace, OpenApiUrlTreeNode currentNode, string declarationName) {
+        var localNameSpace = GetSearchNamespace(currentNode, currentNamespace);
+        return localNameSpace.FindChildByName<ITypeDefinition>(declarationName, false) as CodeElement;
     }
-    private CodeNamespace GetSearchNamespace(bool checkInAllNamespaces, OpenApiUrlTreeNode currentNode, CodeNamespace currentNamespace) {
-        if(checkInAllNamespaces) return rootNamespace;
-        else if (currentNode.DoesNodeBelongToItemSubnamespace()) return currentNamespace.EnsureItemNamespace();
-        else return currentNamespace;
+    private CodeNamespace GetSearchNamespace(OpenApiUrlTreeNode currentNode, CodeNamespace currentNamespace) {
+        if (currentNode.DoesNodeBelongToItemSubnamespace() && !currentNamespace.Name.Contains(modelsNamespace.Name))
+            return currentNamespace.EnsureItemNamespace();
+        else
+            return currentNamespace;
     }
-    private CodeElement AddModelDeclarationIfDoesntExist(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, string declarationName, CodeNamespace currentNamespace, CodeClass inheritsFrom = null, bool checkInAllNamespaces = false) {
```

Comment:
Eliminates the `checkInAllNamespaces` as the `currentNamespace` should already be computed correctly. This also refactors `GetExistingDeclaration` function above.